packages/migrate/src/commands/MigrateDiff.ts

```diff
@@ -244,6 +244,13 @@ ${chalk.bold('Examples')}
         shadowDatabaseUrl: args['--shadow-database-url'],
       })
     } finally {
+      // We need to wait for the "notification(s)" sent by the engine via JSON-RPC to be printed to console.info
+      // In Jest it manifests also as the following if we don't wait:
+      // Cannot log after tests are done. Did you forget to wait for something async in your test?
+      //
+      // we need to return the value in the response then we can remove this
+      await new Promise((resolve) => setTimeout(resolve, 50))
```

Comment:
@tomhoule we really need to clean this up soon and pass the diff in the response 🙈

Reply:
I am pretty sure asynchronously waiting here is not necessary. What we did before is send the notifications and the response on different channels (stderr and stdout), but with the companion PR that is already merged, the print notification is guaranteed to come before the response on stdout, so if the ordering of operations in the CLI matches the ordering of the JSON-RPC messages on stdout, there is no need for extra logic.

If we want to impose more synchronization and delays (e.g. if the log call and the response are reordered by the event loop), we could make the notification a request — the CLI would then send a response (an ACK), and the engine would wait for that before sending its own response. Long term, I think we want to use this mechanism a lot more, not less, because it avoids spreading control flow across both sides of the RPC boundary.

Reply:
Thanks, that makes a lot of sense. Hopefully, this will resolve most of the flakyness we see today :100:

Reply:
@tomhoule the tests fail when we do not wait, see
https://github.com/prisma/prisma/runs/5182960010?check_suite_focus=true